src/Stoa.ts

```diff
@@ -2481,6 +2531,93 @@ class Stoa extends WebService {
             });
     }
 
+    /* Get all proposals
+     * @returns Returns proposals of the ledger.
+     */
+    public async getProposals(req: express.Request, res: express.Response) {
+        const pagination: IPagination = await this.paginate(req, res);
+        this.ledger_storage
+            .getProposals(pagination.pageSize, pagination.page)
+            .then((data: any[]) => {
+                if (data.length === 0) {
+                    return res.status(204).send(`The data does not exist.`);
+                } else {
+                    let proposals: IProposalList[] = [];
+                    for (const row of data) {
+                        proposals.push({
+                            proposal_id: row.proposal_id,
+                            proposal_title: row.proposal_title,
+                            proposal_type: row.proposal_type,
+                            fund_amount: row.fund_amount,
+                            vote_start_height: row.vote_start_height,
+                            vote_end_height: row.vote_end_height,
+                            proposal_status: row.proposal_status,
+                            proposal_date: row.submit_time,
+                            proposer_name: row.proposer_name,
+                            full_count: row.full_count
+                        })
+                    }
+                    return res.status(200).send(JSON.stringify(proposals));
+                }
+            })
+            .catch((err) => {
+                logger.error("Failed to data lookup to the DB: " + err, {
+                    operation: Operation.db,
+                    height: HeightManager.height.toString(),
+                    status: Status.Error,
+                    responseTime: Number(moment().utc().unix() * 1000),
+                });
+                return res.status(500).send("Failed to data lookup");
+            });
+    }
+
+    /* Get proposal by id
+     * @returns Returns proposal of the ledger.
+     */
+    public async getProposalById(req: express.Request, res: express.Response) {
+        const proposal_id: string = String(req.params.proposal_id);
+        if (proposal_id.trim().length === 0) {
+            res.status(400).send(`Parameter proposal id must be set.`);
+            return;
+        }
+        this.ledger_storage
+            .getProposalById(proposal_id)
+            .then((data: any[]) => {
+                if (data.length === 0) {
+                    return res.status(204).send(`The data does not exist.`);
+                } else {
+                    let proposal: IProposalAPI =
+                    {
```

Comment:
app_name,
vote_fee,
proposal_fee_tx_hash,
proposer_address,
proposal_fee_address,
submit_time,
ave_pre_evaluation_score,
pre_evaluation_start_time,
pre_evaluation_end_time,
assess_node_count,
assess_average_score,
assess_completeness_score,
assess_realization_score,
assess_profitability_score,
assess_attractiveness_score,
assess_expansion_score

It needs to add this item.

Reply:
Pre-evaluation data is in separate API, according to the UI design we don't need addition fields